### PR TITLE
Add instructions for 64-bit system support

### PR DESCRIPTION
**Pull request by mhnajafi7**
**Hi There!**
I've made a couple of tweaks that should help iron out some kinks in our project:

- Compiler Setup for 32-bit Systems

I've added instructions to ensure our GNU Toolchain plays nice with 32-bit systems. This means making sure it's set up to work smoothly, especially if you're running a 64-bit system. Just a few commands to enable 32-bit support and install those crucial libraries. It's all about getting your code to compile without a hitch!

- Fixing that Pesky Perl Error

Also, I've dived into the "timeconst.pl" script and made a small but important change. This little tweak should tackle that Perl-related error you've been bumping into during kernel building. Now, instead of checking for defined array values, it's looking for an empty array. It's a small adjustment that should help your build process sail smoothly.

- Testing the Changes

Give these changes a whirl! Follow the steps in the instructions provided for the compiler setup and the script modification. After that, when you're building the kernel, fingers crossed, you shouldn't run into that Perl error anymore. 

- Any Thoughts?

I've tested these changes locally, and they seem to do the trick. They're all about making our lives easier and ensuring your builds are error-free. Would love your thoughts on these tweaks and if they work seamlessly on your end too!

Thanks a bunch! 
Your friendly neighborhood contributor 🌟

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,6 +51,79 @@ Be patient when OpenOCD is flashing. Typically, it takes about 55 seconds.
 Use `make help` to get detailed build targets.
 
 
+# Bug Fixes
```
Comment (jserv) [inline README.md:54]: I am not sure if it makes sense to call it "Bug fixes" since we shall improve the build scripts and the prerequisites first.

**Comment by jserv** [comment]
> Thanks for your feedback and guidance on this. The changes I made were prompted by the older setup for 32-bit compatibility and some faulty scripts. I'm open to your suggestions on how we can better name these modifications to accurately reflect the improvements made. Your guidance in this regard is much appreciated.

You shall refine the subject to address the intention of 32-bit support instead of referring to "bugs."

**Comment by mhnajafi7** [comment]
I've updated the subject name to '64-bit system support.' However, within this section, I explain the necessary steps to modify the 'timeconst.pl' script, aiming to resolve the Perl script-related error.

File: README.md [hunk 2]
```diff
@@ -51,6 +51,79 @@ Be patient when OpenOCD is flashing. Typically, it takes about 55 seconds.
 Use `make help` to get detailed build targets.
 
 
+# 64-bit System Support
```
Comment (jserv) [inline README.md:54]: Should be `##` instead of single `#`.

File: README.md [hunk 3]
```diff
@@ -51,6 +51,79 @@ Be patient when OpenOCD is flashing. Typically, it takes about 55 seconds.
...
+
+## 1. GNU Toolchain
```
Comment (jserv) [inline README.md:56]: Move the parts about GNU Toolchain, compiler, and st-link to section `Prerequisites`.

File: README.md [hunk 4]
```diff
@@ -51,6 +51,79 @@ Be patient when OpenOCD is flashing. Typically, it takes about 55 seconds.
...
+
+## 2. Compiler
+Since the ARM-2010q1 toolchain is intended for 32-bit systems and if your system is 64-bit, you'll need to enable 32-bit support and install necessary libraries. Execute the following commands:
+```bash
+    sudo dpkg --add-architecture i386
+    sudo apt-get update
+    sudo apt-get install libc6:i386 libncurses5:i386 libstdc++6:i386
+```
+
+## 3. ST_Link Programmer
```
Comment (jserv) [inline README.md:86]: Should be `STLINK Tools` as [stlink](https://github.com/stlink-org/stlink) indicates.

**Comment by mhnajafi7** [comment]
I made the necessary changes as you requested. If there is something else, please point it out.

File: README.md [hunk 5]
```diff
@@ -51,6 +51,78 @@ Be patient when OpenOCD is flashing. Typically, it takes about 55 seconds.
...
 
 
+## 64-bit System Support
+Since the ARM-2010q1 toolchain is intended for 32-bit systems and if your system is 64-bit, you'll need to enable 32-bit support and install necessary libraries. Execute the following commands:
+```bash
+    sudo dpkg --add-architecture i386
+    sudo apt-get update
+    sudo apt-get install libc6:i386 libncurses5:i386 libstdc++6:i386
+```
+## Prerequisites
```
Comment (jserv) [inline README.md:61]: Mere the following content into previous `Prerequisites` section.

**Comment by mhnajafi7** [comment]
Done.

**Comment by jserv** [comment]
Thank @mhnajafi7 for contributing!

**Comment by mhnajafi7** [comment]
you're welcome :)

**Pull request merged**